src/server/serverinventorymgr.h

```diff
@@ -0,0 +1,54 @@
+/*
+Minetest
+Copyright (C) 2010-2020 Minetest core development team
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#pragma once
+
+#include "inventorymanager.h"
+#include <functional>
+
+class ServerEnvironment;
+
+class ServerInventoryMgr : public InventoryManager
+{
+public:
+	ServerInventoryMgr(ServerEnvironment *env);
+	virtual ~ServerInventoryMgr();
+	/*
+		Shall be called with the environment and the connection locked.
+	*/
+	Inventory *getInventory(const InventoryLocation &loc);
+	void setInventoryModified(const InventoryLocation &loc);
+
+	// Creates or resets inventory
+	Inventory *createDetachedInventory(const std::string &name, IItemDefManager *idef,
+			const std::string &player = "");
+	bool removeDetachedInventory(const std::string &name);
+
+	void sendDetachedInventories(const std::string &peer_name, bool incremental,
+			std::function<void(const std::string &, Inventory *)> apply_cb);
+	const std::string &getAttachedPlayer(const std::string &name) const;
+
+private:
+	ServerEnvironment *m_env = nullptr;
+
+	// key = name
+	std::map<std::string, Inventory *> m_detached_inventories;
```

Comment:
This should rather be a `std::unordered_map<std::string, DetachedInventory>`
with 
```cpp
struct DetachedInventory {
  Inventory *inv;
  std::string players; // "" means visible to all players
};
```

This avoids doing a second lookup in `sendDetachedInventories` and also allows `getAttachedPlayer` to be removed.

Reply:
i will take a look to see how we can optimize that, but your comment sounds to increase more again the quality on this code part (and it's why spliting big objects is good !!)

Reply:
i don't remember why this map passed outside of my radar on the c++11 move, anyway let's go